lib/chef/provider/mount/windows.rb

```diff
@@ -41,8 +41,10 @@ def initialize(new_resource, run_context)
         def load_current_resource
           if is_volume(@new_resource.device)
             @mount = Chef::Util::Windows::Volume.new(@new_resource.name)
-          else #assume network drive
+          elsif @new_resource.name.include?(":") && !@new_resource.mount_point #assume network drive
             @mount = Chef::Util::Windows::NetUse.new(@new_resource.name)
```

Comment:
This would also be mount_point not name

Reply:
I thought it similar but to make it backward compatible I committed this change. Anyways I will update this and make it `name` to `mount_point`.